### PR TITLE
fix #8837 feat(nimbus): Do localization validation only when localized

### LINKED ISSUE #8837: Localizations validation occurring for non localized experiment
An experiment is failing the review serializer with a warning that there's an error on the Overview page, but nothing is showing.  Looking at the gql response we see:

image

however there should be no validation for this field since the experiment is not localized:

image

Enabling the checkbox does indeed show the error:

image

This validation shouldn't kick in unless the experiment is localized.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3491)

### PR DESCRIPTION
**Pull request by brennie**
Becuase:
- the localizations field is only considered in the case of localized experiments

this commit:
- only validates the localizations field for localized experiments.

### CODE REVIEW

File: experimenter/experimenter/experiments/api/v5/serializers.py [hunk 1]
```diff
@@ -1652,39 +1631,51 @@ def _validate_localizations(self, data):
...
                 {"locales": ["Locales must not be empty for a localized experiment."]}
             )
 
-        experiment_locale_codes = [locale.code for locale in experiment_locales]
-        localizations = json.loads(localizations_json)
+        localizations_json = data.get("localizations")
+        try:
+            localizations = json.loads(localizations_json)
+        except Exception as e:
+            raise serializers.ValidationError(f"Invalid JSON: {e}")
```
Comment (jrbenny35) [inline experimenter/experimenter/experiments/api/v5/serializers.py:1644]: Can you catch a ValidationError only? If not, I'd suggest using ```from e``` so we don't lose the context of the actual error.

**Review by jrbenny35 — COMMENTED** [review]
Just had a thought.

File: experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py [hunk 2]
```diff
@@ -2113,6 +2111,27 @@ def test_localized_value_missing_keys(self, l10n_obj, error_msg):
             {"reference_branch": {"feature_value": [error_msg]}},
         )
 
+    def test_not_localized_with_localizations(self):
```
Comment (eliserichards) [inline experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py:2114]: Wonderful! 🎉

**Review by eliserichards — APPROVED** [review]

**Review by yashikakhurana — APPROVED** [review]
localization validated ❗ 🚀

File: experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py (hunk 2 above)
Comment (jaredlockhart) [reply experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_ready_for_review_serializer.py:2114]: Excellent 💯

**Pull request merged**